### DISCUSSION TITLE
Hyprland site

### DISCUSSION THREAD

**Discussion by System-x64** [hyprwm/Hyprland#430] [General]
### Heyloo its me again...ik i am really annoying

Soo...as the title says it all I created this sorta sample home page for "Hyprland" 's official site and i just want to ask (if you guys allow ofc) can i like build the entire site with the install and configure page so that you guys could have an official website as such? 

Actually the thing is, in about a time of 2 years from now i might be applying to many colleges in the US so by making this site and help an ambitious project like this... i could really write all about this in my essays and stuff basically helping you will greatly help me aswell 

Hopefully you'll like this idea and reply with a positive response (i mean its ok if you dont, i understand it takes alot of time and resources, but i am ready to put all of my brain cells into this!
Cheer!

![screenshot_20220726_175108](https://user-images.githubusercontent.com/90208727/180777393-ae16f6b6-9a9f-4254-812e-d0b41859a118.png)
*this is still a very early design, i would change  many of the things as this site grows

**Comment by vaxerski** [comment]
Yes, that's fine (I say fine because adding a fifth domain to my yearly expenses is, oh well, thats just 10$ i guess) 

But my numero uno question: why the hell react????? For a simple webpage?! 

React is massive bloat and my puny server hosts like 4 websites at once, I don't want it murdered. If you take a look at my website at https://vaxry.net/ or blog at https://blog.vaxry.net/  they are built with express.js.

If you cannot provide any reasonable explanation as to why react and not something either lightweight or simply better, maybe a simple host like express.js and a pure js website would be better? We do not plan on adding any social media there...

**Reply by NotAShelf** [reply]
My personal reason for choosing react and nextjs is that they are like bringing an entire toolbox when you need a hammer and a screwdriver. Allows for ultimately more potential to be fulfilled and you often end up with something that is much better than your original plan, as the rabbit hole goes pretty deep.

Though I suggest *not* using react if it's a **simple** webpage, like a onepager. otherwise it's one of the best, if not, the best framework for a full-scale, in-depth website.

**Reply by vaxerski** [reply]
Explain how. I've built my blog with express.js without many issues.

You're allowed to use videos, images, and (woo, scary!) code.

Also, hyprland is NOT a WM. It's a Wayland compositor.

(Remember I am not a web dev and have relatively little experience outside node.js and HTML/CSS/pure js)

**Reply by System-x64** [reply]
Well i am fine with both toh I always prefer React for its ability to divide code into different components rather than having a hugh mess of divs spans and links in one long file, it just makes it easier to debug or change small stuff, secondly, by chance if we ever need to apply javascript into the site ( like for making the site mobile friendly) even toh it's possible in vanilla js, its really a pain to use all the DOMs and getElementBy stuffs when you can directly inject js between the jsx tags in react but considering you host all sites on one single server...i let this decision be up to you weather to use react or vanilla js.

**Reply by vaxerski** [reply]
I'd prefer vanilla js (anything that can be run from node.js for the backend is ok)

Remember that we are not planning to make this the best site on the interwebz. It's just to be a nice, small, clean looking webpage.

**Reply by System-x64** [reply]
Ok, vanilla js then...imma start building

**Reply by fufexan** [reply]
Just chiming in: do we *really* need JS? CSS can probably do whatever we need (animations, mobile friendliness, etc).

**Reply by vaxerski** [reply]
same thing as I thought, but some things are just easier to do in JS and I for example am often lazy.

> mobile friendliness

as if I care, lol

**Reply by godalming123** [reply]
What about svelte and sveltekit it can do almost everything react can but is quite a lot lighter weight

**Reply by vaxerski** [reply]
but... why??? it's a static website.

**Reply by godalming123** [reply]
I agree theres no reason to use overly-complicated solutions

**Comment by System-x64** [comment]
ok, almost a week now but atleast the home page is compeleted...be sure to tell me any edits you want to have

https://github.com/System-x64/hyprland-site (edited)

**Reply by vaxerski** [reply]
why not a git repo???

**Reply by System-x64** [reply]
I actually tried to but my repo is kinda messed up, ill fix that later but till then please bear with me 😓

**Reply by System-x64** [reply]
Ok, its up on the repo as well now

https://github.com/System-x64/hyprland-site

**Reply by System-x64** [reply]
umm..if everything's good, can I start working on the install page?

**Reply by vaxerski** [reply]
yeah lgtm

**Reply by System-x64** [reply]
Alrighty, Finally it is complete! check it out!

https://github.com/System-x64/hyprland-site

**Reply by System-x64** [reply]
Tell me if you need any changes, or we are all ready to host it, from now on it entirely depends on you, however just handle the SEO part to make sure it happens shows when searching for Hyprland

**Reply by System-x64** [reply]
just tell me when you host it, cant wait to see it live!

**Reply by System-x64** [reply]
Umm...is their anything you want me to change @vaxerski ?

**Reply by vaxerski** [reply]
oh, sorry, for some reason i didnt get any notifs. Let me check.

**Reply by vaxerski** [reply]
Looks pretty good, I can take over from here on. Needs some changes, but mostly substantive, not graphical. I'll let you know when it's public.

**Reply by System-x64** [reply]
K, a huge thank you for letting me be a part of it!

**Comment by godalming123** [comment]
Have started working on [this](https://github.com/godalming123/hyprland-website) for ~1 hour it use vanilla html css and js

**Reply by System-x64** [reply]
Sry bud, but I have already completed the site, just doing the back-end stuff now

**Reply by godalming123** [reply]
But vaxry doesn't want to use react

**Reply by System-x64** [reply]
Ik, we have discussed this already, i've created it on express (lmao forgot to rename the title)

**Reply by godalming123** [reply]
Ok will use these styles for other sites though so still a good use of time

**Comment by godalming123** [comment]
A demo is available [here](https://godalming123.github.io/hyprland-website/)

**Comment by vaxerski** [comment]
The website is now live on [hyprland.org](https://hyprland.org/), but it's still very early, it's just been up since today, so expect some changes.

The all new and revamped wiki, that still is nowhere near finished is at [wiki.hyprland.org](https://wiki.hyprland.org/)

**Reply by System-x64** [reply]
ayyyy finally....also just realized, opening this on my ipad broke the entire home page padding if possible could you please upload the site's code with all the changes you've made?

**Reply by vaxerski** [reply]
i will...


soon:tm:

**Reply by System-x64** [reply]
my thx

**Reply by vaxerski** [reply]
https://github.com/hyprwm/hyprland-website

**Reply by System-x64** [reply]
Progress....I'm kinda done with the heading and navbar part, now just need to work on the features and the footer

Also till then could you maybe just work on the SEO part becoz yk....whats the point of all this headache if people can't even see this in their search results 

![Screenshot (13)](https://user-images.githubusercontent.com/90208727/184679368-140383b2-165d-4fe2-8518-dbb37263bb52.png)

**Reply by vaxerski** [reply]
Sir I am a C++ programmer, wdym SEO, I can't even talk to a woman, how could I be good at advertising

**Reply by NotAShelf** [reply]
Just add the keywords you want people to find Hyprland by.
"wayland"
"wayland compositor" 
"hyprland"
etc.
so people find the website when they search for those terms.

P.S. Hey that's my rice in the image.

**Reply by System-x64** [reply]
Lmao...well just watch some 5 min videos on SEOs, you'll be gud to go...we dont want to be the top most search on searching wayland, we just want it to appear on searching Hyprland

_plus its easy_

**Reply by System-x64** [reply]
> Just add the keywords you want people to find Hyprland by. "wayland" "wayland compositor" "hyprland" etc. so people find the website when they search for those terms.
> 
> P.S. Hey that's my rice in the image.

yeye thats it, no biggo deal

**Reply by System-x64** [reply]
> Just add the keywords you want people to find Hyprland by. "wayland" "wayland compositor" "hyprland" etc. so people find the website when they search for those terms.
> 
> P.S. Hey that's my rice in the image.

Oh so u are "Rafela", had hard time finding the owner of this rice :>

**Reply by NotAShelf** [reply]
> Oh so u are "Rafela", had hard time finding the owner of this rice :>

That is, in fact, me.

**Reply by System-x64** [reply]
> > Oh so u are "Rafela", had hard time finding the owner of this rice :>
> 
> That is, in fact, me.

So is that your real name?

**Reply by NotAShelf** [reply]
> > > Oh so u are "Rafela", had hard time finding the owner of this rice :>
> > 
> > 
> > That is, in fact, me.
> 
> So is that your real name?

Unfortunately, yes. I go by raf most of the time.

**Reply by System-x64** [reply]
> > > > Oh so u are "Rafela", had hard time finding the owner of this rice :>
> > > 
> > > 
> > > That is, in fact, me.
> > 
> > 
> > So is that your real name?
> 
> Unfortunately, yes. I go by raf most of the time.

well then....hi there _raf_ ;-)

**Reply by System-x64** [reply]
@vaxerski  Ok, i've completely made the site mobile friendly, the new code can be found in a pull request in the hyprland-website repo.

**Comment by System-x64** [comment]
God!!!! Why this annoying svg doesn't get scaled up ![image](https://user-images.githubusercontent.com/90208727/185756521-7499c283-0165-490e-8868-373965c8277f.png)

**Reply by System-x64** [reply]
i mean it looks perfectly fine on desktop 
![image](https://user-images.githubusercontent.com/90208727/185756567-1587acaf-48fc-4f50-b145-17437a5948fb.png)
